client/src/components/CalculateMenu/CalculateMenu.js

```diff
@@ -48,56 +56,59 @@ export const CalculateMenu = ({
   }
   return (
     <>
-      <Box
-        p={4}
-        borderRadius="lg"
-        m={4}
-        bgColor="white"
-        shadow="base"
-        zIndex="1"
-        style={{ position: "absolute", width: "600px", top: 0, right: 0 }}
-      >
-        <HStack spacing={2} justifyContent="space-between">
-          <Box flexGrow={1}>
+      <Box sx={{ flexGrow: 1 }} zIndex="1" className={styles.container}>
+        <Grid
+          container
+          spacing={{ xs: 1, md: 1 }}
+          columns={{ xs: 2, sm: 2, md: 8 }}
+        >
+          <Grid item xs={2} sm={4} md={4}>
             <Autocomplete>
-              <Input
+              <TextField
+                sx={{ width: "90%" }}
                 type="text"
                 placeholder="Origin"
-                ref={originRef}
-                value={centerAddress}
+                variant="standard"
+                value={origin}
+                onChange={(e) => setOrigin(e.target.value)}
```

Comment:
can you create `onChangeHandler` function with useCallback, instead of this will be `onChange={onChangeHandler}`